packages/authorization-utils/src/domain.rs

```diff
@@ -11,54 +11,88 @@ pub enum Domain {
 pub struct ExternalDomain {
     pub name: String,
     pub execution_environment: ExecutionEnvironment,
-    pub connector: Connector,
     pub processor: String,
-    pub callback_proxy: CallbackProxy,
 }
 
-impl ExternalDomain {
+#[cw_serde]
+pub enum ExecutionEnvironment {
+    Cosmwasm(CosmwasmBridge),
+    Evm(Encoder, EvmBridge),
+}
+
+impl ExecutionEnvironment {
     pub fn get_connector_address(&self) -> Addr {
-        match &self.connector {
-            Connector::PolytoneNote { address, .. } => address.clone(),
+        match self {
+            ExecutionEnvironment::Cosmwasm(CosmwasmBridge::Polytone(polytone_info)) => {
+                polytone_info.polytone_note.address.clone()
+            }
+            ExecutionEnvironment::Evm(_, EvmBridge::Hyperlane(hyperlane_info)) => {
+                hyperlane_info.mailbox.clone()
+            }
         }
     }
 
-    pub fn get_callback_proxy_address(&self) -> Addr {
-        match &self.callback_proxy {
-            CallbackProxy::PolytoneProxy(address) => address.clone(),
+    pub fn get_callback_address(&self) -> Addr {
+        match self {
+            ExecutionEnvironment::Cosmwasm(CosmwasmBridge::Polytone(polytone_info)) => {
+                polytone_info.polytone_proxy.clone()
+            }
+            ExecutionEnvironment::Evm(_, EvmBridge::Hyperlane(hyperlane_info)) => {
+                hyperlane_info.mailbox.clone()
+            }
         }
     }
 
-    pub fn get_connector_state(&self) -> PolytoneProxyState {
-        match &self.connector {
-            Connector::PolytoneNote { state, .. } => state.clone(),
+    pub fn get_polytone_proxy_state(&self) -> Option<PolytoneProxyState> {
+        match self {
+            ExecutionEnvironment::Cosmwasm(CosmwasmBridge::Polytone(polytone_info)) => {
+                Some(polytone_info.polytone_note.state.clone())
+            }
+            ExecutionEnvironment::Evm(_, _) => None,
         }
     }
```

Comment:
I feel like having this under `impl ExecutionEnvironment` makes it a little confusing, what about we move it to `CosmwasmBridge` or `PolytoneConnectors`?  seems like it's only called under `PolytoneCallbackMsg::CreateProxy` handler so we already know we're dealing with polytone there so shouldn't be much issues